### ISSUE TITLE
Quantile LightGBM - inconsistent deciles

### ISSUE THREAD

**Issue by TomekPro** [microsoft/LightGBM#3448]
Hi, I'm doing Quantile LightGBM, defined as follows:
```
QuantileEstimator(lgb.LGBMRegressor(n_jobs=-1,
                             seed=1234,
                             learning_rate=0.1,
                             reg_sqrt=True,
                             objective = 'quantile',
                             n_estimators=100))
```

Where QuantileEstimator is just:
```
class QuantileEstimator(BaseEstimator, RegressorMixin):

    def __init__(self, model):
        """
        """
        self.alphas = [round(x, 1) for x in np.arange(0.1, 1, 0.1)]
        self.model_factory = []
        self.model = model
        super().__init__()

    def fit(self, X, y=None):
        
        for a in self.alphas:
            model_i = clone(self.model)
            model_i = model_i.set_params(**{'alpha': a})
            model_fitted = model_i.fit(X, y)
            self.model_factory.append(model_fitted)
        return self

    def predict(self, X):
        predictions = pd.DataFrame()
        for a, m in zip(self.alphas, self.model_factory):
            predictions["y_pred_" + str(a)] = m.predict(X)
        return predictions
```

Sometimes it happens that following decile for individual predictiions has lower value than the previous one.
![image](https://user-images.githubusercontent.com/72195667/95588201-66e06280-0a43-11eb-9c3e-acf78a7f016a.png)

Any idea why that's happening? I understand that those are different models, but still as I set the same seed I would expect that the results for different deciles should be monotonic.

**Comment by StrikerRUS** [comment]
Duplicate of #3447.
